crates/zksync/core/src/vm/tracer.rs

```diff
@@ -310,8 +310,10 @@ impl<S: Send, H: HistoryMode> DynTracer<S, SimpleMemory<H>> for CheatcodeTracer
                     self.call_context.block_hashes.get(&block_number.to_ru256())
                 {
                     self.farcall_handler.set_immediate_return(block_hash.to_vec());
-                    return;
+                } else {
+                    self.farcall_handler.set_immediate_return(rU256::ZERO.to_be_bytes_vec());
```

Comment:
Could there be a case where block_number  in call data yields a value that's not the current block but also some block number that's not in the current block hashes map (say for example asking for the hash of a block earlier than 256 blocks ago or getting something weird due to corrupted calldata)? If so it will return 0 as well, Is this fine?

Reply:
Hey fede, according to the documentation when asking the blockhash of either the current block or any other that is not one of the 256 previous ones the return value should be zero

> blockhash(uint blockNumber) returns (bytes32): hash of the given block when blocknumber is one of the 256 most recent blocks; otherwise returns zero

https://docs.soliditylang.org/en/latest/units-and-global-variables.html